### PR TITLE
GRN - reference revision and tests

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
GRN - reference revision and tests

GRN is a special case of NormalizeL2 (across channel) with the `bias` as `eps` 
so the reference is a simple call of `reference::normalize_l2` (no changes needed)

[runtime/reference/grn.hpp](https://github.com/mitruska/openvino/blob/3d5670ebdf13be0f63f182c5cc77cbba7b9c55e9/ngraph/core/reference/include/ngraph/runtime/reference/grn.hpp#L15-L19)
```
void grn(const T* data, T* out, float bias, const Shape& data_shape)
{
    normalize_l2(data, out, data_shape, {1}, bias, op::EpsMode::ADD);
}
```

In this PR:

- [x] Add shared single layer GRN tests instance for cpu
- [x] Add new template reference GRN tests (and remove legacy backend GRN fused op tests)

### Tickets:
 - 44630 (45772, 45772)

### CODE REVIEW

File: docs/template_plugin/tests/functional/op_reference/grn.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+using namespace InferenceEngine;
+
+struct GrnParams {
+    template <class IT, class OT>
+    GrnParams(const float bias, const ngraph::PartialShape& shape, const ngraph::element::Type& iType, const ngraph::element::Type& oType,
+              const std::vector<IT>& iValues, const std::vector<OT>& oValues, size_t iSize = 0, size_t oSize = 0)
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/grn.cpp:21]: As I see you don't use `iSize`, `oSize` prease remove redundant arguments

File: docs/template_plugin/tests/functional/op_reference/grn.cpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+TEST_P(ReferenceGrnLayerTest, CompareWithHardcodedRefs) {
+    Exec();
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    smoke_GRN_With_Hardcoded_Refs, ReferenceGrnLayerTest,
+    ::testing::Values(
+        // bias 1e-6 // 2D // 3D // 4D
+        GrnParams(1e-6, ngraph::PartialShape {3, 4}, ngraph::element::f32, ngraph::element::f32, std::vector<float> {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12},
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/grn.cpp:65]: Should we extend these tests for other data types?

**Review by jdanieck — COMMENTED** [review]
I a missing two things:
- serialization SLT
- adding GRN to trusted ops

**Comment by mitruska** [comment]
> I a missing two things:
> 
> * serialization SLT
> * adding GRN to trusted ops

Done

File: docs/template_plugin/tests/functional/op_reference/grn.cpp (hunk 2 above)
Comment (mitruska) [reply docs/template_plugin/tests/functional/op_reference/grn.cpp]: Refactored to generate params with different types (bf16, f16, f32), please take a look.

File: docs/template_plugin/tests/functional/op_reference/grn.cpp (hunk 1 above)
Comment (mitruska) [reply docs/template_plugin/tests/functional/op_reference/grn.cpp]: Done, also `oType` is redundant for GRN so I've removed it too.

**Review by jdanieck — APPROVED** [review]

**Comment by ilyachur** [comment]
@mitruska Please rerun CI

**Review by ilyachur — APPROVED** [review]

**Pull request merged**